.circleci/config.yml

```diff
@@ -1,13 +1,13 @@
 version: 2.1
 
 executors:
-  circle-jdk23-executor:
+  jdk24-executor:
     working_directory: ~/micrometer
     environment:
       GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError"'
     resource_class: medium+
     docker:
-      - image: cimg/openjdk:23.0.2
+      - image: gradle:8-jdk24
```

Comment:
I'm thinking if this works, we should migrate all of our images so that they are consistent.

Reply:
Consistency is a good goal. I was trying to change as little as possible and try out the new image without potentially breaking everything and needing to revert everything. If the trend continues of the CircleCI JDK convenience images not doing new releases, we'll also need to move off them to keep testing with the latest JDK patch releases.